encoder/vaapiencoder_hevc.h

```diff
@@ -0,0 +1,178 @@
+
+/*
+ *  vaapiencoder_hevc.h - hevc encoder for va
+ *
+ *  Copyright (C) 2014-2015 Intel Corporation
+ *    Author: Wind Yuan <feng.yuan@intel.com>
+ *    Author: Xu Guangxin <guangxin.xu@intel.com>
+ *    Author: Li Zhong <zhong.li@intel.com>
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public License
+ *  as published by the Free Software Foundation; either version 2.1
+ *  of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301 USA
+ */
+
+#ifndef vaapiencoder_hevc_h
+#define vaapiencoder_hevc_h
+
+#include "vaapiencoder_base.h"
+#include "vaapi/vaapiptrs.h"
+#include "common/lock.h"
+#include <list>
+#include <queue>
+#include <deque>
+#include <pthread.h>
+#include <va/va_enc_hevc.h>
+
+namespace YamiMediaCodec{
+class VaapiEncPictureHEVC;
+class VaapiEncoderHEVCRef;
+class VaapiEncStreamHeaderHEVC;
+
+typedef struct shortRFS
```

Comment:
could we change this to a nested class of  VaapiEncoderHEVC. and typedef is not needed